synapse/lib/scrape.py

```diff
@@ -81,6 +81,17 @@ def cve_check(match: regex.Match):
     valu = s_chop.replaceUnicodeDashes(valu)
     return valu, cbfo
 
+# Reference NIST 7695
+# Common Platform Enumeration: Naming Specification Version 2.3
+# Figure 6-3. ABNF for Formatted String Binding
+_cpe23_regex = r'''
+# Capture the valu group
+(?P<valu>cpe:2\.3:[aho\*-]
+(?::([a-z0-9-_.*?]|\\[\\?*!"#$%&\'()*+,/;:<=>@\[\]^`{|}~])+){5}
```

Comment:
i think the optional spec_chrs matches need to be split out to avoid matching too many unescaped *, like `cpe:2.3:a:*:******:*:*:*:*:*:*:*:*`

Reply:
also * is in the list of chars twice

Reply:
i think something like `(?::([*-]|[?*]?([a-z0-9-_.]|\\[\\?*!"#$%&\'()+,\/;:<=>@\[\]^`{|}~])+[?*]?)){5}` would be correct

Reply:
actually spec_chrs allows multiple ? so ```(?::([*-]|(\?*|\*)?([a-z0-9-_.]|\\[\\?*!"#$%&\'()+,\/;:<=>@\[\]^`{|}~])+(\?*|\*)?)){5}```

Reply:
@Cisphyx Thanks for this feedback! I realize i misread the ABNF in at least one place, so I'll revisit that.

Reply:
@Cisphyx I rewrote the regex to reflect the ANBF And retain the ordering that it described in the reference document.